### ISSUE TITLE
##[error]The nuget command failed with exit code(1) and error(An item with the same key has already been added.

### ISSUE THREAD

**Issue by troyjudd14** [NuGet/Home#13456]
### NuGet Product Used

NuGet.exe

### Product Version

6.10

### Worked before?

6.9

### Impact

I'm unable to use this version

### Repro Steps & Context

In our yaml file we restore NuGet packages from our own feed and from NuGet.org. 
We are specifying Version 6.9.1 as a workaround.

![image](https://github.com/NuGet/Home/assets/147873029/81890c3c-f4f9-4e53-ade9-dc4b1a7e6807)

steps:
- task: NuGetToolInstaller@1

- task: NuGetCommand@2
  displayName: "Restore NuGet package dependencies from custom name feed."
  inputs:
    command: 'restore'
    restoreSolution: '$(solution)'
    feedsToUse: 'select'
    vstsFeed: 'our feed key'

- task: NuGetCommand@2
  displayName: "Restore NuGet package dependencies from NuGet.org feed."
  inputs:
    command: 'restore'
    restoreSolution: '**/*.sln'
    feedsToUse: 'select'

### Verbose Logs

```shell
2024-05-21T19:19:32.0057280Z ##[section]Starting: Restore NuGet package dependencies from feed.
2024-05-21T19:19:32.0067250Z ==============================================================================
2024-05-21T19:19:32.0067384Z Task         : NuGet
2024-05-21T19:19:32.0067449Z Description  : Restore, pack, or push NuGet packages, or run a NuGet command. Supports NuGet.org and authenticated feeds like Azure Artifacts and MyGet. Uses NuGet.exe and works with .NET Framework apps. For .NET Core and .NET Standard apps, use the .NET Core task.
2024-05-21T19:19:32.0067720Z Version      : 2.238.1
2024-05-21T19:19:32.0067779Z Author       : Microsoft Corporation
2024-05-21T19:19:32.0067850Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/package/nuget
2024-05-21T19:19:32.0067952Z ==============================================================================
2024-05-21T19:19:33.0007158Z [command]C:\Windows\system32\chcp.com 65001
2024-05-21T19:19:33.0964601Z Active code page: 65001
2024-05-21T19:19:33.4640790Z Detected NuGet version 6.10.0.107 / 6.10.0+6970ffe10efff51d43862ea8f8a2ef0170250262.6970ffe10efff51d43862ea8f8a2ef0170250262
2024-05-21T19:19:33.5217240Z [command]C:\hostedtoolcache\windows\NuGet\6.10.0\x64\nuget.exe sources Add -NonInteractive -Name eb5226ba-ad7e-4568-ab21-ae78862f0a66 -Source https://pkgs.dev.azure.com/repo-name/feed-key-part-1/_packaging/feed-key-part-2/nuget/v3/index.json -ConfigFile D:\a\1\Nuget\tempNuGet_1007.config
2024-05-21T19:19:35.7583675Z Package source with Name: eb5226ba-ad7e-4568-ab21-ae78862f0a66 added successfully.
2024-05-21T19:19:35.7685511Z [command]C:\hostedtoolcache\windows\NuGet\6.10.0\x64\nuget.exe sources Add -NonInteractive -Name NuGetOrg -Source https://api.nuget.org/v3/index.json -ConfigFile D:\a\1\Nuget\tempNuGet_1007.config
2024-05-21T19:19:36.6715155Z Package source with Name: NuGetOrg added successfully.
2024-05-21T19:19:36.6814512Z [command]C:\hostedtoolcache\windows\NuGet\6.10.0\x64\nuget.exe restore D:\a\1\s\Portal.sln -Verbosity Detailed -NonInteractive -ConfigFile D:\a\1\Nuget\tempNuGet_1007.config
2024-05-21T19:19:54.1074961Z An item with the same key has already been added.
2024-05-21T19:19:54.1075287Z System.ArgumentException: An item with the same key has already been added.
2024-05-21T19:19:54.1075522Z    at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
2024-05-21T19:19:54.1075787Z    at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
2024-05-21T19:19:54.1076057Z    at NuGet.CommandLine.RestoreCommand.GetPackagesConfigToProjectPath(PackageRestoreInputs packageRestoreInputs)
2024-05-21T19:19:54.1076340Z    at NuGet.CommandLine.RestoreCommand.<PerformNuGetV2RestoreAsync>d__59.MoveNext()
2024-05-21T19:19:54.1076573Z --- End of stack trace from previous location where exception was thrown ---
2024-05-21T19:19:54.1076811Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2024-05-21T19:19:54.1077080Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2024-05-21T19:19:54.1077340Z    at NuGet.CommandLine.RestoreCommand.<ExecuteCommandAsync>d__52.MoveNext()
2024-05-21T19:19:54.1077587Z --- End of stack trace from previous location where exception was thrown ---
2024-05-21T19:19:54.1077808Z    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2024-05-21T19:19:54.1078068Z    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2024-05-21T19:19:54.1078287Z    at NuGet.CommandLine.Command.Execute()
2024-05-21T19:19:54.1078509Z    at NuGet.CommandLine.Program.MainCore(String workingDirectory, String[] args)
2024-05-21T19:19:54.1078701Z NuGet Version: 6.10.0.107
2024-05-21T19:19:54.1079022Z MSBuild auto-detection: using msbuild version '16.11.2.50704' from 'C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
2024-05-21T19:19:54.1081182Z WARNING: Project file D:\a\1\s\Portal\ cannot be found.
2024-05-21T19:19:54.1081359Z MSBuild P2P timeout [ms]: 120000
2024-05-21T19:19:54.1082148Z C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\bin\msbuild.exe "C:\Users\VssAdministrator\AppData\Local\Temp\NuGetScratch\z4wkm3zx.x40.nugetinputs.targets" /t:GenerateRestoreGraphFile /nologo /nr:false /v:q /p:NuGetPropsFile="C:\Users\VssAdministrator\AppData\Local\Temp\NuGetScratch\qmszf1y4.ohy.nugetrestore.props" /p:NuGetRestoreTargets="C:\Users\VssAdministrator\AppData\Local\Temp\NuGetScratch\10k25o4w.k4a.nugetrestore.targets" /p:RestoreUseCustomAfterTargets="True" /p:DisableCheckingDuplicateNuGetItems="True" /p:RestoreTaskAssemblyFile="C:\hostedtoolcache\windows\NuGet\6.10.0\x64\nuget.exe" /p:RestoreSolutionDirectory="D:\a\1\s\\" /p:RestoreConfigFile="D:\a\1\Nuget\tempNuGet_1007.config" /p:SolutionDir="D:\a\1\s\\" /p:SolutionName="Portal"
2024-05-21T19:19:54.1082780Z 
2024-05-21T19:19:54.1847615Z ##[error]The nuget command failed with exit code(1) and error(An item with the same key has already been added.
System.ArgumentException: An item with the same key has already been added.
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at NuGet.CommandLine.RestoreCommand.GetPackagesConfigToProjectPath(PackageRestoreInputs packageRestoreInputs)
   at NuGet.CommandLine.RestoreCommand.<PerformNuGetV2RestoreAsync>d__59.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.CommandLine.RestoreCommand.<ExecuteCommandAsync>d__52.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.CommandLine.Command.Execute()
   at NuGet.CommandLine.Program.MainCore(String workingDirectory, String[] args))
2024-05-21T19:19:54.1858650Z ##[error]Packages failed to restore
2024-05-21T19:19:54.1868463Z ##[section]Finishing: Restore NuGet package dependencies from  feed.
```

**Comment by ncook-hxgn** [comment]
This is affecting at least all of Production Software at Hexagon MI

**Comment by nkolev92** [comment]
Hey all, 

thanks for filing this and apologies for the inconvenience. 
Can you help us get a repro so we can add an appropriate test case?

It seems like NuGet somehow thinks a specific packages.config path applies to more than 1 project.
Do you set the PackagesConfigPath project property in any of your projects? 
Do you maybe have 2 csproj files in 1 folder?

A repro sln would be ideal, but I understand that might not be possible.

**Comment by nkolev92** [comment]
Another quick thing. 

If all of you that are facing this issue could upgrade your Visual Studio version as well, can you please check if you're hitting a similar issue there?

**Comment by troyjudd14** [comment]
Hey @nkolev92, 
I'll do my best to assist. Our repo consists of a precompiled website supported with 3 projects. One cs project and two vb projects. I am using Visual Studio Professional 2022 locally and don't have any issues. From the log file it looks like ADO is using Enterprise 2019. I'm not sure if I can upgrade that. I'll look into it.

**Comment by troyjudd14** [comment]
Update: In ADO, I updated the yaml's vmImage to use Windows 2022 which uses Visual Studio 2022 and received the same error.

**Comment by nkolev92** [comment]
My instructions weren't clear enough, my bad.

Context: When a new version of NuGet is release, there's components that share code in Visual Studio, NuGet.exe and .NET SDK. 
17.10 of Visual Studio, matches 6.10 of NuGet.exe, matches 8.0.3xx of .NET SDK.


1. If you load the solution in Visual Studio and attempt to restore, do you get the same error?


2. If VisualStudio is running 17.10, can you try in VS and see if you're hitting an issue there? I don't expect that to be the case. 
Another workaround is also, if you are using an updated version of Visual Studio, you can use https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#restoring-packagereference-and-packagesconfig-projects-with-msbuild.

3. If I gave you a locally built NuGet.exe, would you be willing to give that a shot and provide me some of the logs?
I can link the branch I'll build it from.

**Comment by troyjudd14** [comment]
Thank you for the clarification. I am updating Visual Studio to 17.10 right now. I'll get back to you on whether or not I get the error once it finishes installing.

**Comment by nkolev92** [comment]
After a further investigation, I think msbuild /t:restore /p:RestorePackagesConfig='true" and  VS restore shouldn't have this problem. 

It looks like it's only NuGet.exe.

**Comment by Nevcairiel** [comment]
I'm facing a similar issue with VS17.10 built-in NuGet support, the msbuild command worked as a work-around however.

**Comment by troyjudd14** [comment]
@nkolev92 I finished updating VS to 17.10. I use the package manager built into Visual Studio so I don't use the NuGet CLI but I placed the current version of the CLI in the same directory as my solution and ran nuget restore from the command line editor. I'm getting the same error "An item with the same key has already been added." 
![image](https://github.com/NuGet/Home/assets/147873029/9b6d7656-7196-4bcd-9e66-6d2d3246708f)
Let me know if you need me to run a locally built version of NuGet.exe.

**Comment by nkolev92** [comment]
When you open the sln in VisualStudio and run restore, do you get any errors then?

Can you please try running with: [NuGet.zip](https://github.com/NuGet/Home/files/15424719/NuGet.zip)

The branch is: https://github.com/NuGet/NuGet.Client/pull/5818. 

It'll tell me enough whether my hunch is right, or if I need to do more "printf debugging". 

I really appreciate your help in this @troyjudd14

**Comment by troyjudd14** [comment]
I get an error when running the new version. 
`Unhandled Exception: System.IO.FileLoadException: Could not load file or assembly 'NuGet, Version=6.11.0.32767, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. Strong name validation failed. (Exception from HRESULT: 0x8013141A) ---> System.Security.SecurityException: Strong name validation failed. (Exception from HRESULT: 0x8013141A)
   --- End of inner exception stack trace ---`

**Comment by nkolev92** [comment]
Try: It's a test-signed build. 
If that doesn't work, I'll get an officially signed build.
[NuGet.zip](https://github.com/NuGet/Home/files/15424873/NuGet.zip)

**Comment by troyjudd14** [comment]
I tried the build you just posted in the zip file and I get the same error. Is there something different I should do for a test-signed build? Also, I need to sign off for the evening. I'll pick this up again tomorrow.

**Comment by nkolev92** [comment]
You'd need to disable strong name validation. 
I'll drop a real signed build here soon, and you can test it out tomorrow.

**Comment by nkolev92** [comment]
[NuGet.zip](https://github.com/NuGet/Home/files/15438107/NuGet.zip)
@troyjudd14 and others, can you please try this and provide the logs. 

Thanks.

**Comment by troyjudd14** [comment]
@nkolev92 This version seems to have worked for me locally. I'd like to test it in an ADO pipeline. Is this version available to use in ADO?

PS C:\Users\ah98019\ADO Repo\portal> nuget restore
MSBuild auto-detection: using msbuild version '17.10.4.21802' from 'C:\Program Files\Microsoft Visual Studio\2022\Professional\MSBuild\Current\Bin'.
WARNING: Project file C:\Users\ah98019\ADO Repo\portal\Portal\ cannot be found.
WARNING: Multiple projects are associated with one 'packages.config' file. This will lead to NuGetAudit configuration not being respected in the second project. It is recommended that each project has their own packages.config file. packages.config: 'C:\Users\ah98019\ADO Repo\portal\Apps\packages.config', projects: 'C:\Users\ah98019\ADO Repo\portal\Apps\ExternalCLib.csproj', 'C:\Users\ah98019\ADO Repo\portal\Apps\ExternalVLib.vbproj'.
All packages listed in packages.config are already installed.
  GET https://api.nuget.org/v3/vulnerabilities/index.json
  OK https://api.nuget.org/v3/vulnerabilities/index.json 52ms
  GET https://api.nuget.org/v3-vulnerabilities/2024.05.23.23.36.45/vulnerability.base.json
  GET https://api.nuget.org/v3-vulnerabilities/2024.05.23.23.36.45/2024.05.24.05.36.45/vulnerability.update.json
  OK https://api.nuget.org/v3-vulnerabilities/2024.05.23.23.36.45/vulnerability.base.json 106ms
  OK https://api.nuget.org/v3-vulnerabilities/2024.05.23.23.36.45/2024.05.24.05.36.45/vulnerability.update.json 233ms
    [CredentialProvider]VstsCredentialProvider - Acquired bearer token using 'MSAL Interactive'
    [CredentialProvider]VstsCredentialProvider - Attempting to exchange the bearer token for an Azure DevOps session token.

**Comment by nkolev92** [comment]
Thank you for confirming that.

* There's no way of getting that in AzDO, since it's just a build that's not published anywhere. 
* My hunch was correct, there are 2 projects in the same folder. 
'C:\Users\ah98019\ADO Repo\portal\Apps\packages.config', projects: 'C:\Users\ah98019\ADO Repo\portal\Apps\ExternalCLib.csproj', 'C:\Users\ah98019\ADO Repo\portal\Apps\ExternalVLib.vbproj', 2 projects are in C:\Users\ah98019\ADO Repo\portal\Apps. 

We'll get this fixed. 

A couple paths forward for unblocking yourself:

* I'd encourage not having multiple projects in the same directory.  There's obj/bin sharing, NuGet won't know how to build the project graph since the dependencies could be conflicting. Pretty confident the tooling won't allow you to do this automatically, and it's something that requires manual work today.
* Switch to using `msbuild -t:restore -p:RestorePackagesConfig="true". This would mean you don't need nuget.exe anymore. As long as all of your packages.config files are associated with a project, this will work. The functionality is the same.
* Wait for the fix to be released. 

Call to action: 

* If *anyone* sees this problem, but does not have 2 projects in the same directory, please let us know. That'd mean there's other scenarios we're not aware of.

**Comment by Nevcairiel** [comment]
I'm facing the same generic error message in VS17.10, msbuild or nuget.exe v6.10 actually work fine - only VS17.10 integrated fails. We do not have multiple projects in the directory (unless its somehow counting the linux Makefile next to the .vcxproj, but I doubt that?)

I realize this is a very generic error message. I got this out of the Package Manager console, if it helps any:

```
PM> Update-Package -reinstall
Update-Package : An item with the same key has already been added.
At line:1 char:1
+ Update-Package -reinstall
+ ~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Update-Package], ArgumentException
    + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.UpdatePackageCommand
 
PM> $error[0].Exception.StackTrace
   at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at NuGet.PackageManagement.PackageRestoreManager.<GetRestoreAuditProperties>d__31.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.PackageRestoreManager.<RestoreMissingPackagesAsync>d__35.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.PowerShellCmdlets.NuGetPowerShellBaseCommand.<CheckMissingPackagesAsync>d__64.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.PackageManagement.PowerShellCmdlets.PackageActionBaseCommand.<<Preprocess>b__39_0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at NuGet.PackageManagement.PowerShellCmdlets.PackageActionBaseCommand.Preprocess()
   at NuGet.PackageManagement.PowerShellCmdlets.UpdatePackageCommand.Preprocess()
   at NuGet.PackageManagement.PowerShellCmdlets.UpdatePackageCommand.ProcessRecordCore()
   at NuGet.PackageManagement.PowerShellCmdlets.NuGetPowerShellBaseCommand.ProcessRecord()
```

I can also take this to a new issue if its distinctly separate.

**Comment by nkolev92** [comment]
> I'm facing the same generic error message in VS17.10, msbuild or nuget.exe v6.10 actually work fine - only VS17.10 integrated fails. We do not have multiple projects in the directory (unless its somehow counting the linux Makefile next to the .vcxproj, but I doubt that?)
> 
> I realize this is a very generic error message. I got this out of the Package Manager console, if it helps any:
> 
> ```
> PM> Update-Package -reinstall
> Update-Package : An item with the same key has already been added.
> At line:1 char:1
> + Update-Package -reinstall
> + ~~~~~~~~~~~~~~~~~~~~~~~~~
>     + CategoryInfo          : NotSpecified: (:) [Update-Package], ArgumentException
>     + FullyQualifiedErrorId : NuGetCmdletUnhandledException,NuGet.PackageManagement.PowerShellCmdlets.UpdatePackageCommand
>  
> PM> $error[0].Exception.StackTrace
>    at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
>    at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
>    at NuGet.PackageManagement.PackageRestoreManager.<GetRestoreAuditProperties>d__31.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at NuGet.PackageManagement.PackageRestoreManager.<RestoreMissingPackagesAsync>d__35.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at NuGet.PackageManagement.PowerShellCmdlets.NuGetPowerShellBaseCommand.<CheckMissingPackagesAsync>d__64.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at NuGet.PackageManagement.PowerShellCmdlets.PackageActionBaseCommand.<<Preprocess>b__39_0>d.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
>    at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
>    at NuGet.PackageManagement.PowerShellCmdlets.PackageActionBaseCommand.Preprocess()
>    at NuGet.PackageManagement.PowerShellCmdlets.UpdatePackageCommand.Preprocess()
>    at NuGet.PackageManagement.PowerShellCmdlets.UpdatePackageCommand.ProcessRecordCore()
>    at NuGet.PackageManagement.PowerShellCmdlets.NuGetPowerShellBaseCommand.ProcessRecord()
> ```
> 
> I can also take this to a new issue if its distinctly separate.

Any chance I can get repro of that?

It's related to the same feature/assumption, but it's in a different codepath from the one here, so we can track a separate issue for that.

**Comment by Nevcairiel** [comment]
> Any chance I can get repro of that?
> 
> It's related to the same feature/assumption, but it's in a different codepath from the one here, so we can track a separate issue for that.

I'll try to strip down the project and solution to a minimal case and open a new issue over or after the weekend.

**Comment by bstordrup** [comment]
> [NuGet.zip](https://github.com/NuGet/Home/files/15438107/NuGet.zip) @troyjudd14 and others, can you please try this and provide the logs.
> 
> Thanks.

@nkolev92, I had the same key problem also after updating nuget.exe to latest version. The NuGet.zip that you supplied, resolved my issue, and it confirmed the packages.config being shared issue.

```
C:\Git\Visma\VismaBusiness\VismaBusiness>BuildScripts\bin\nuget restore Visma.BusinessServices.sln
MSBuild auto-detection: using msbuild version '17.9.8.16306' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\bin'.
WARNING: Multiple projects are associated with one `packages.config` file. This will lead to NuGetAudit configuration not being respected in the second project. It is recommended that each project has their own packages.config file. packages.config: 'C:\Git\Visma\VismaBusiness\VismaBusiness\Visma.BusinessHost\packages.config', projects: 'C:\Git\Visma\VismaBusiness\VismaBusiness\Visma.BusinessHost\Visma.BusinessHost.csproj', 'C:\Git\Visma\VismaBusiness\VismaBusiness\Visma.BusinessHost\Visma.BusinessHost64.csproj'.
WARNING: Multiple projects are associated with one `packages.config` file. This will lead to NuGetAudit configuration not being respected in the second project. It is recommended that each project has their own packages.config file. packages.config: 'C:\Git\Visma\VismaBusiness\VismaBusiness\Visma.DebugHost\packages.config', projects: 'C:\Git\Visma\VismaBusiness\VismaBusiness\Visma.DebugHost\Visma.DebugHost.csproj', 'C:\Git\Visma\VismaBusiness\VismaBusiness\Visma.DebugHost\Visma.DebugHost64.csproj'.
```

**Comment by Seltzer** [comment]
I recently upgraded VS2022 to 17.10.0 and a similar error started appearing for me when trying to do anything in NuGet at the solution level. 

After excluding various projects from the solution to narrow it down, I discovered that it was choking on two projects which have the same name but different locations.

**Comment by Nevcairiel** [comment]
> I recently upgraded VS2022 to 17.10.0 and a similar error started appearing for me when trying to do anything in NuGet at the solution level.
> 
> After excluding various projects from the solution to narrow it down, I discovered that it was choking on two projects which have the same name but different locations.

I can confirm that this is the same issue for us. Two projects with the same name, different folder - neither of which actually uses NuGet packages.

**Comment by david-wimmer** [comment]
> > I recently upgraded VS2022 to 17.10.0 and a similar error started appearing for me when trying to do anything in NuGet at the solution level.
> > After excluding various projects from the solution to narrow it down, I discovered that it was choking on two projects which have the same name but different locations.
> 
> I can confirm that this is the same issue for us. Two projects with the same name, different folder - neither of which actually uses NuGet packages.

Same issue here.

**Comment by Sonix** [comment]
I am seeing the issue with two .vcxproj files in the same folder. Will pin to Nuget version 6.9.1 for now.

**Comment by nkolev92** [comment]
* I have created a tracking issue for the VS/PMC side - https://github.com/NuGet/Home/issues/13465
* I have a PR out for fixing NuGet.exe - https://github.com/NuGet/NuGet.Client/pull/5818

**Comment by nkolev92** [comment]
Whoever's hitting the Visual Studio issue, I need some help a repro, since Visual Studio disallows creating/loading the same project name twice in Visual Studio. 

I'm unable to get a repro to load even  by manual editing.
I'd appreciate if you can zip up a solution that does that. 

Tagging folks that reported that issue: @david-wimmer @Nevcairiel

**Comment by devAlex-zz** [comment]
@nkolev92 You can have a project with the same name in different folders. **Important**: You need non sdk style projects to reproduce the problem. I made a demo repo for you https://github.com/devAlex-zz/Nuget13456

**Comment by ghost** [comment]
Hi all, just adding to this, we have the same issue with two vcxproj files in the same directory, even though only one of them uses a nuget package.

**Comment by nkolev92** [comment]
Thanks @devAlex-zz!

@william-bourne 
packages.config works based off of convention, so NuGet is probably detecting the packages.config for both projects.
If you open the PM UI of the project that doesn't use a NuGet package, the packages listed in the packages.config are going to be shown as installed despite the fact that no references exist in the project file.

**Comment by nkolev92** [comment]
* PR for fixing VS/PMC -  https://github.com/NuGet/NuGet.Client/pull/5825 - this fix has some assumptions, but the proper change caries a bigger risk, so we'll get that taken care of in a minor version instead of servicing.
* PR for fixing NuGet.exe - https://github.com/NuGet/NuGet.Client/pull/5818
